### PR TITLE
ASYNC_start_job: more readable documentation for handling ASYNC* APIs

### PR DESCRIPTION
**Pull request by tomato42**
Recently I was trying to figure out how to use the OpenSSL APIs combined with the ASYNC multithreading and found the documentation hard to follow.

Added few more comments that would probably made it easier to the past me.

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

File: doc/man3/ASYNC_start_job.pod [hunk 1]
```diff
@@ -250,17 +261,23 @@ The following example demonstrates how to use most of the core async APIs:
...
-     /* Return control back to main */
+     /*
+      * Return control back to main just before calling a blocking
+      * method. The main thread will wait until pipefds[0] is ready
+      * for reading before returning control to this thread.
+      */
      ASYNC_pause_job();
 
-     /* Clear the wake signal */
+     /* Perform the blocking call */
```
Comment (mattcaswell) [inline doc/man3/ASYNC_start_job.pod:280]: This change doesn't seem correct. This shouldn't ever block. It just clearing the wake signal that we wrote with the `write` call a few lines above.

File: doc/man3/ASYNC_start_job.pod [hunk 2]
```diff
@@ -311,6 +330,8 @@ The following example demonstrates how to use most of the core async APIs:
          ASYNC_WAIT_CTX_get_all_fds(ctx, &waitfd, &numfds);
          FD_ZERO(&waitfdset);
          FD_SET(waitfd, &waitfdset);
+
+         /* Wait for the job to be ready for wakeup */
```
Comment (mattcaswell) [inline doc/man3/ASYNC_start_job.pod:334]: In this demo we know we will never really wait because `jobfunc` above immediately set the wait signal on `waitfd` rather than waiting for some external event to occur.

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: doc/man3/ASYNC_start_job.pod (hunk 1 above)
Comment (tomato42) [reply doc/man3/ASYNC_start_job.pod:280]: in this example code, yes, it won't; but then this example code does nothing.
In real world code, this is the blocking call we want to protect against by use of `ASYNC_pause_job()`.

File: doc/man3/ASYNC_start_job.pod (hunk 2 above)
Comment (tomato42) [reply doc/man3/ASYNC_start_job.pod:334]: That's precisely the issue with the documentation. As a user I need examples that explain what happens when I use real world code. Not examples that show how to burn cycles doing nothing.

**Comment by tomato42** [comment]
About the failing CI check:
```
doc/man3/ASYNC_start_job.pod:1: Missing man section number (likely, 3) in L<select(2)>
doc/man3/ASYNC_start_job.pod:1: Missing man section number (likely, 3) in L<poll(2)>
doc/man3/ASYNC_start_job.pod:1: reference to non-existing select(2)
doc/man3/ASYNC_start_job.pod:1: reference to non-existing poll(2)
```
we can't link to basic C man pages?

**Comment by nhorman** [comment]
According to this:
https://perldoc.perl.org/perlpod#Formatting-Codes

We should be able to.  I wonder if successful linking is incumbent on the target man pages being installed?  I would imagine that the ci images that do the doc-nits check have them all stripped

**Comment by nhorman** [comment]
Oh, I found the problem, its twofold:

1) find-doc-nits only checks for man pages in sections 1,3,5 and 7.  We need to add section 2 to the check
2) find-doc-nits assumes all links resolve to files in the openssl repo (excluding hyperlinks and such).  since unix man pages don't we would need to find some way to make them appear in the tree, have find-doc-nits understand them as nroff pages (rather than pod files), and not processes them

**Comment by tomato42** [comment]
Sounds like I shouldn't link to them then...

**Comment by nhorman** [comment]
yeah, probably not, as the L<> tag assumes that the link target is included in the docs tree.  We could potentially use this syntax:
```
L<select|http://external.link.to/man/page/you/want/to/target
```

There are several sites that provide canonical man pages in html form, but I'm not sure what the policy is on linking to external sources

**Comment by tomato42** [comment]
[no-shared-macos (macos-14)](https://github.com/openssl/openssl/actions/runs/8425345866/job/23071230348?pr=23941#logs) failure looks unrelated: seems to be a race condition in the test case

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by nhorman** [comment]
merged, thank you for your contribution
